From 92832c1ae4d9bec38dd356e43c1dd96e68bbb5e1 Mon Sep 17 00:00:00 2001
From: Brad Ebinger <breadley@google.com>
Date: Wed, 6 May 2020 16:18:12 -0700
Subject: [PATCH 1/3] Specify the component for the ACTION_SHOW_ECM_EXIT_DIALOG
 Intent

Specify the component for the ACTION_SHOW_ECM_EXIT_DIALOG intent
so it can not be intercepted by a malicious application and modified.

Bug: 155094269
Test: manually trigger ECBM dialog and exit.
Merged-In: Id5df0b428bd7e24aa11fb9dd4c920d1019f1b389
Change-Id: Id5df0b428bd7e24aa11fb9dd4c920d1019f1b389
(cherry picked from commit 11c41d321d0c2dc3631f37d2f8f3ebc745f454d2)
---
 src/com/android/phone/EmergencyCallbackModeService.java | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/com/android/phone/EmergencyCallbackModeService.java b/src/com/android/phone/EmergencyCallbackModeService.java
index a07f7aa93..86f8fe607 100644
--- a/src/com/android/phone/EmergencyCallbackModeService.java
+++ b/src/com/android/phone/EmergencyCallbackModeService.java
@@ -189,8 +189,10 @@ public class EmergencyCallbackModeService extends Service {
 
         // PendingIntent to launch Emergency Callback Mode Exit activity if the user selects
         // this notification
-        PendingIntent contentIntent = PendingIntent.getActivity(this, 0,
-                new Intent(EmergencyCallbackModeExitDialog.ACTION_SHOW_ECM_EXIT_DIALOG), 0);
+        Intent intent = new Intent(this, EmergencyCallbackModeExitDialog.class);
+        intent.setAction(EmergencyCallbackModeExitDialog.ACTION_SHOW_ECM_EXIT_DIALOG);
+        PendingIntent contentIntent = PendingIntent.getActivity(this, 0, intent,
+                PendingIntent.FLAG_IMMUTABLE);
         builder.setContentIntent(contentIntent);
 
         // Format notification string
-- 
2.17.1

